src/types/profile.js

```diff
@@ -20,9 +20,36 @@ export type resourceTypeEnum = number;
 export type ThreadIndex = number;
 
 /**
- * The stack table is the minimal representation of a call stack. Each stack entry
- * consists of the frame at the top of the stack, and the prefix for the stack that
- * came before it. Stacks can be shared between samples.
+ * The stack table stores the tree of stack nodes of a thread.
+ * The shape of the tree is encoded in the prefix column: Root stack nodes have
+ * null as their prefix, and every non-root stack has the stack index of its
+ * "caller" / "parent" as its prefix.
+ * Every stack node also has a frame and a category.
+ * A "call stack" is a list of frames. Every stack index in the stack table
+ * represents such a call stack; the "list of frames" is obtained by walking
+ * the path in the tree from the root to the given stack node.
+ *
+ * Stacks are used in the thread's samples; each sample refers to a stack index.
+ * Stacks can be shared between samples.
+ *
+ * With this representation, every sample only needs to store a single integer
+ * to identify the sample's stack.
+ * We take advantage of the fact that many call stacks in the profile have a
+ * shared prefix; storing these stacks as a tree saves a lot of space compared
+ * to storing them as actual lists of frames.
+ *
+ * The category of a stack node is always non-null and is derived from a stack's
+ * frame and its prefix. Frames can have null categories, stacks cannot. If a
+ * stack's frame has a null category, the stack inherits the category of its
+ * prefix stack. Root stacks whose frame has a null stack have their category
+ * set to the "default category". (The default category is currently defined as
+ * the category in the profile's category list whose color is "grey", and such
+ * a category is required to be present.)
```

Comment:
we should enforce this at processing time: if no such category exist, we should create it. (cf other comments about the same subject)